src/geo/transform.js

```diff
@@ -283,8 +314,10 @@ class Transform {
         const coord0 = [p.x, p.y, 0, 1];
         const coord1 = [p.x, p.y, 1, 1];
 
-        vec4.transformMat4(coord0, coord0, this.pixelMatrixInverse);
-        vec4.transformMat4(coord1, coord1, this.pixelMatrixInverse);
+        const pixelMatrixInverse = this._calcMatrices(viewport);
```

Comment:
This operation is expensive enough that we will need to cache `pixelMatrixInverse` for at least the global viewport.